tests/scripts/identifierAnalysisByType.fsx

```diff
@@ -0,0 +1,152 @@
+    eprintfn ""
+    eprintfn "Sample output is at https://gist.github.com/dsyme/abfa11bebf0713251418906d55c08804"
+
+//let projectFile = Path.Combine(__SOURCE_DIRECTORY__, @"..\..\src\Compiler\FSharp.Compiler.Service.fsproj")
+//let projectFile = Path.Combine(__SOURCE_DIRECTORY__, @"..\..\src\FSharp.Build\FSharp.Build.fsproj")
+let projectFile = Path.GetFullPath(argv[1])
+
+let cwd = System.Environment.CurrentDirectory |> System.IO.DirectoryInfo
+
+let _toolsPath = Init.init cwd None
+
+printfn "Cracking project options...."
+let opts =
+    match ProjectLoader.getProjectInfo projectFile [] BinaryLogGeneration.Off [] with 
+    | Result.Ok res -> res
+    | Result.Error err -> failwithf "%s" err
+
+let checker = FSharpChecker.Create()
+
+let checkerOpts = checker.GetProjectOptionsFromCommandLineArgs(projectFile, [| yield! opts.SourceFiles; yield! opts.OtherOptions  |] )
+
+printfn "Checking project...."
+let results = checker.ParseAndCheckProject(checkerOpts) |> Async.RunSynchronously
+
+printfn "Grouping symbol uses...."
+let symbols = results.GetAllUsesOfAllSymbols()
+
+let rec stripTy (ty: FSharpType) = 
+    if ty.IsAbbreviation then stripTy ty.AbbreviatedType else ty
+
+let getTypeText (sym: FSharpMemberOrFunctionOrValue) =
+    let ty = stripTy sym.FullType
+    FSharpType.Prettify(ty).Format(FSharpDisplayContext.Empty)
+
+symbols
+|> Array.choose (fun vUse -> match vUse.Symbol with :? FSharpMemberOrFunctionOrValue as v -> Some (v, vUse.Range) | _ -> None)
+|> Array.filter (fun (v, _) -> v.GenericParameters.Count = 0)
+|> Array.filter (fun (v, _) -> v.CurriedParameterGroups.Count = 0)
+|> Array.filter (fun (v, _) -> not v.FullType.IsGenericParameter)
+|> Array.map (fun (v, vUse) -> getTypeText v, v, vUse)
+|> Array.filter (fun (vTypeText, v, _) -> 
+    match vTypeText with 
+    | "System.String" -> false
+    | "System.Boolean" -> false
+    | "System.Int32" -> false
+    | "System.Int64" -> false
+    | "System.Object" -> false
+    | "Microsoft.FSharp.Collections.List<Microsoft.FSharp.Core.string>" -> false
+    | "Microsoft.FSharp.Core.Option<Microsoft.FSharp.Core.string>" -> false
+    | s when s.EndsWith(" Microsoft.FSharp.Core.[]") -> false // for now filter array types
+    | _ when v.DisplayName.StartsWith "_" -> false
+    | _ -> true)
+|> Array.groupBy (fun (vTypeText, _, _) -> vTypeText)
+|> Array.map (fun (key, g) ->
+    key, 
+    (g 
+     |> Array.groupBy (fun (_, v, _) -> v.DisplayName)
+     |> Array.sortByDescending (snd >> Array.length)))
+|> Array.filter (fun (_, g) -> g.Length > 1)
+|> Array.sortByDescending (fun (key, g) -> Array.length g)
+|> Array.iter (fun (key, g) -> 
+    let key = key.Replace("Microsoft.FSharp", "FSharp").Replace("FSharp.Core.", "")
+    printfn "Type: %s" key
+    for (nm, entries) in g do
+       printfn "    %s (%d times)" nm (Array.length entries)
+       for (_, _, vUse) in entries do
+           printfn "        %s" (vUse.ToString())
+    printfn "")
+
+(*
```

Comment:
Should this be deleted?

Reply:
Probably.  I hacked it across from https://github.com/fsprojects/FSharp.Formatting/blob/main/src/fsdocs-tool/ProjectCracker.fs and I think it was all a workaround for something, I'm not sure.  Or else maybe it was there to cope with non-standard install locations for `dotnet`